src/Orleans.Autofac/AutofacOrleansDependencyResolver.cs

```diff
@@ -0,0 +1,129 @@
+﻿/*
+Project Orleans Cloud Service SDK ver. 1.0
+ 
+Copyright (c) Microsoft Corporation
+ 
+All rights reserved.
+ 
+MIT License
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and 
+associated documentation files (the ""Software""), to deal in the Software without restriction,
+including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
+subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS
+OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+using System;
+using System.Security;
+using Autofac;
+using Autofac.Core.Lifetime;
+using Orleans.Runtime;
+
+namespace Orleans.Autofac
+{
+    [SecurityCritical]
+    public class AutofacOrleansDependencyResolver : IDependencyResolver
+    {
+        private bool _disposed;
+        readonly ILifetimeScope _container;
+        readonly IDependencyScope _rootDependencyScope;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AutofacOrleansDependencyResolver"/> class.
+        /// </summary>
+        /// <param name="container">The container that nested lifetime scopes will be create from.</param>
+        public AutofacOrleansDependencyResolver(ILifetimeScope container)
```

Comment:
With that naming I just followed the naming from the Autofac WebApi adapter, which was AutofacWebApiDependencyResolver, but I will remove that.